test/torchaudio_unittest/transforms/sox_compatibility_test.py

```diff
@@ -61,3 +64,26 @@ def test_vad(self, filename):
         data, sample_rate = load_wav(path)
         result = T.Vad(sample_rate)(data)
         self.assert_sox_effect(result, path, ['vad'])
+
+    def test_vad_warning(self):
```

Comment:
Can you provide the docstring which describes the condition when the warning should be issued and when not, so that any future maintainer can understand the intend easily?

I think we can merge `test_vad_warning` and `test_vad_no_warning`.